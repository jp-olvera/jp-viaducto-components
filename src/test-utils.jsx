import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
