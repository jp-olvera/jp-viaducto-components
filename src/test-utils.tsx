import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
