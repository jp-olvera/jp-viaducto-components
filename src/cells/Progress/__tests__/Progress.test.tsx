/* eslint-env jest */

import React from 'react';
import { render, screen } from '../../../test-utils';
import {
  Progress, CircleLoader, ProgressBar, StepLoader,
} from '..';

import '@testing-library/jest-dom';

describe('Progress component', () => {
  test('should have circle rendered', () => {
    const { getByTestId, container } = render(
      <Progress totalSteps={3} completedSteps={0} currentStep={0} />,
    );
    expect(getByTestId('loader')).toBeInTheDocument();
    expect(container.querySelector('circle')).toBeInTheDocument();
  });
  test('should have steps loader rendered', () => {
    const { getByTestId } = render(
      <Progress
        loader='steps'
        totalSteps={3}
        completedSteps={0}
        currentStep={0}
      />,
    );
    expect(getByTestId('loader')).toBeInTheDocument();
  });
  test('should have progress bar rendered', () => {
    const { getByTestId, container } = render(
      <Progress
        loader='progress'
        totalSteps={3}
        completedSteps={0}
        currentStep={0}
      />,
    );
    expect(getByTestId('loader')).toBeInTheDocument();
    expect(
      container.querySelector('.progress-bar-container'),
    ).toBeInTheDocument();
  });
  test('should have equal values', () => {
    const { container } = render(
      <Progress
        loader='progress'
        totalSteps={3}
        completedSteps={3}
        currentStep={3}
      />,
    );
    expect(container).toBeInTheDocument();
  });
  test('should render default props', () => {
    const { container } = render(<Progress />);
    expect(container).toBeInTheDocument();
  });
});

describe('StepLoader component', () => {
  test('should have add current step', () => {
    const total = 3;

    const { rerender } = render(
      <StepLoader completed={0} totalSteps={total} />,
    );
    expect(screen.queryByText('0 / 3')).not.toBeNull();
    rerender(<StepLoader completed={3} totalSteps={total} />);
    expect(screen.getByText('3 / 3')).not.toBeNull();
  });
});

describe('<CircleLoader/>', () => {
  test('should render Circle Loader and create Snapshot', () => {
    render(<CircleLoader />);
    expect(screen.getByTestId('loader')).toMatchSnapshot();
  });

  test('should render Circle Loader with props and create Snapshot', () => {
    render(
      <CircleLoader
        strokeWidth={2}
        circumference={5}
        actualProgress={1}
        currentProgress={1}
        radius={10}
      />,
    );
    expect(screen.getByTestId('loader')).toMatchSnapshot();
  });
});

describe('<ProgressBar/>', () => {
  test('should render ProgressBar and create snapshot', () => {
    render(<ProgressBar totalSteps={50} completedSteps={51} />);
    expect(screen.getByTestId('loader')).toMatchSnapshot();
  });
  test('should render ProgressBar and create snapshot with props', () => {
    render(<ProgressBar totalSteps={10} completedSteps={5} />);
    expect(screen.getByTestId('loader')).toMatchSnapshot();
  });
});

describe('<StepLoader/>', () => {
  test('should render StepLoader and create snapshot', () => {
    render(<StepLoader totalSteps={14} completed={16} finishTextColor='red' />);
    expect(screen.getByTestId('loader')).toMatchSnapshot();
  });
  test('should render StepLoader and create snapshot with props', () => {
    render(<StepLoader totalSteps={15} completed={7} color='red' />);
    expect(screen.getByTestId('loader')).toMatchSnapshot();
  });
});
