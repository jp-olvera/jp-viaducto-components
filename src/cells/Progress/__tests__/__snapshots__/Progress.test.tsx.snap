// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleLoader/> should render Circle Loader and create Snapshot 1`] = `
<div
  class="StyleLoader__StyledLoader-sc-1bymfut-0 kZnXVG"
  data-testid="loader"
>
  <svg
    height="25px"
    width="25px"
  >
    <circle
      cx="12.5"
      cy="12.5"
      fill="transparent"
      r="0"
      stroke="#CECECE"
      stroke-width="0"
    />
    <circle
      class="progress-ring__circle"
      cx="12.5"
      cy="12.5"
      fill="transparent"
      r="0"
      stroke="#937B3D65"
      stroke-dasharray="0 0"
      stroke-width="0"
      style="stroke-dashoffset: 0;"
    />
    <circle
      class="progress-ring__circle"
      cx="12.5"
      cy="12.5"
      fill="transparent"
      r="0"
      stroke="#937B3D"
      stroke-dasharray="0 0"
      stroke-width="0"
      style="stroke-dashoffset: 0;"
    />
  </svg>
</div>
`;

exports[`<CircleLoader/> should render Circle Loader with props and create Snapshot 1`] = `
<div
  class="StyleLoader__StyledLoader-sc-1bymfut-0 kZnXVG"
  data-testid="loader"
>
  <svg
    height="25px"
    width="25px"
  >
    <circle
      cx="12.5"
      cy="12.5"
      fill="transparent"
      r="10"
      stroke="#CECECE"
      stroke-width="2"
    />
    <circle
      class="progress-ring__circle"
      cx="12.5"
      cy="12.5"
      fill="transparent"
      r="10"
      stroke="#937B3D65"
      stroke-dasharray="5 5"
      stroke-width="2"
      style="stroke-dashoffset: 1;"
    />
    <circle
      class="progress-ring__circle"
      cx="12.5"
      cy="12.5"
      fill="transparent"
      r="10"
      stroke="#937B3D"
      stroke-dasharray="5 5"
      stroke-width="2"
      style="stroke-dashoffset: 1;"
    />
  </svg>
</div>
`;

exports[`<ProgressBar/> should render ProgressBar and create snapshot 1`] = `
<div
  class="StyleLoader__StyledProgressBar-sc-1bymfut-2 cCSQrl"
  color="#937B3D"
  data-testid="loader"
  max="50"
  value="50"
>
  <div
    class="progress-bar-container"
  >
    <div
      class="progress-bar stripes animated reverse slower"
    >
      <span
        class="progress-bar-inner"
      />
    </div>
  </div>
</div>
`;

exports[`<ProgressBar/> should render ProgressBar and create snapshot with props 1`] = `
<div
  class="StyleLoader__StyledProgressBar-sc-1bymfut-2 deMAeB"
  color="#937B3D"
  data-testid="loader"
  max="10"
  value="5"
>
  <div
    class="progress-bar-container"
  >
    <div
      class="progress-bar stripes animated reverse slower"
    >
      <span
        class="progress-bar-inner"
      />
    </div>
  </div>
</div>
`;

exports[`<StepLoader/> should render StepLoader and create snapshot 1`] = `
<h1
  class="StyleLoader__StyledStepProgress-sc-1bymfut-1 uLVF"
  data-testid="loader"
>
  14 / 14
</h1>
`;

exports[`<StepLoader/> should render StepLoader and create snapshot with props 1`] = `
<h1
  class="StyleLoader__StyledStepProgress-sc-1bymfut-1 ilwoBs"
  color="red"
  data-testid="loader"
>
  7 / 15
</h1>
`;
